### PR TITLE
Add resource pools

### PR DESCRIPTION
**Pull request by skatsaounis**
Enhance `gomaasclient` with the ability to manage resource pool objects.

### CODE REVIEW

**Comment by skatsaounis** [comment]
Resolves: #25

File: client/resource_pool.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+	"github.com/google/go-querystring/query"
+	"github.com/maas/gomaasclient/entity"
+)
+
+type ResourcePool struct {
+	ApiClient ApiClient
+}
+
+func (r *ResourcePool) client(id int) ApiClient {
+	return r.ApiClient.GetSubObject(fmt.Sprintf("resourcepools/%v", id))
```
Comment (troyanov) [inline client/resource_pool.go:17]: According to https://maas.io/docs/api the endpoint to Get/Update/Delete should be in singular form?

> DELETE /MAAS/api/2.0/resourcepool/{id}/
> GET /MAAS/api/2.0/resourcepool/{id}/
> PUT /MAAS/api/2.0/resourcepool/{id}/

```suggestion
	return r.ApiClient.GetSubObject(fmt.Sprintf("resourcepool/%v", id))
```

**Review by troyanov — CHANGES_REQUESTED** [review]
LGTM but with a small issue.

File: client/resource_pool.go (hunk 1 above)
Comment (skatsaounis) [reply client/resource_pool.go]: nice catch! Fixed

**Review by troyanov — APPROVED** [review]

**Pull request merged**